src/pages/FarmingPage.js

```diff
@@ -9,11 +9,71 @@ import {
 import { PageWrapper, FullWrapper } from "../components";
 import { RowBetween } from "../components/Row";
 import Search from "../components/Search";
+import DropdownSelect from "../components/DropdownSelect";
 import { useMedia } from "react-use";
 import FarmingList from "../components/FarmingList";
+import memoize from "../utils/memoize";
 
 function FarmingPage() {
+  // UNIX time
+  const timeNow = Math.floor(Date.now() / 1000);
+
   const campaignData = useLiqudityMiningCampaignData();
+  const [campaigns, setCampaigns] = useState(campaignData);
+  const [campaignStatus, setCampaignStatus] = useState("active");
+
+  useEffect(() => {
+    if (campaignData) {
+      filterCampaigns("active");
+    }
+  }, [campaignData]);
+
+  const filterCampaigns = (status) => {
+    if (!campaignData) { return };
+    if (status === "active") {
+      const filtered = memoizedActiveCampaigns();
+      setCampaigns(filtered);
+    } else if (status === "expired") {
+      const filtered = memoizedExpiredCampaigns();
+      setCampaigns(filtered);
+    }
+    setCampaignStatus(status);
+  }
+
+  const activeCampaigns = () => {
+    return Object.keys(campaignData)
+      .filter(key => campaignData[key].startsAt < timeNow && campaignData[key].endsAt > timeNow)
+      .reduce((obj, key) => {
+        return {
+          ...obj,
+          [key]: campaignData[key]
+        };
+      }, {});
+  }
+
+  const expiredCampaigns = () => {
+    return Object.keys(campaignData)
+      .filter(key => campaignData[key].endsAt < timeNow)
+      .reduce((obj, key) => {
+        return {
+          ...obj,
+          [key]: campaignData[key]
+        };
+      }, {});
+  }
+
+  const options = {
+    "expired": "Expired Campaigns",
+    "active": "Active Campaigns"
+  }
+
+  const handleFilterCampaigns = (selected) => {
+    const status = Object.keys(options).find(key => options[key] === selected);
+    filterCampaigns(status);
+  }
+
+  const memoizedActiveCampaigns = memoize(activeCampaigns);
+  const memoizedExpiredCampaigns = memoize(expiredCampaigns);
```

Comment:
1) I think this filtering should be done in useLiqudityMiningCampaignData and served directly to component since its only use case for given hook
2) Subgraph query needs to be changed to include expired campaigns since it just fetches active campaigns
![Screenshot 2021-11-12 at 18 32 56](https://user-images.githubusercontent.com/33226956/141509876-cfe37cad-08ae-4e24-97a1-8522c6ca43c7.png)